src/vos/tests/csum_extent_tests.c

```diff
@@ -1010,7 +1010,16 @@ csum_fault_injection_multiple_extents_tests(void **state)
 				 csum_count_per_extent);
 		assert_int_equal(get_csum_total(&test, fetch_extents),
 				csums_count_total);
-		assert_memory_not_equal(test.update_csum_buf,
+		/* It looks is there is random issues in setting
```

Comment:
I'm not familiar enough with the fault injection mechanisms used, but seems like we should understand why there are random issues. I believe fault injection does have the ability to be random in injecting faults. Perhaps it's enabled?

Reply:
I faced the same problem when I used DAOS_FAIL_ONCE flag and noticed the fault injection flag was not set based on debug print information (with DAOS_FAIL_ONCE, I was able to reproduce the problem on frequently). After I enabled the DAOS_FAIL_ALWAYS, I am not able to reproduce problem locally. I have created DAOS-3015 for someone to take a look at fault injection issue. Presently, I think we should not cause random CI failures due to this flaky problem. Any thoughts?